tools/delete_mri_upload.pl

```diff
@@ -0,0 +1,578 @@
+	push(@files, $archiveLocation);
+	
+	foreach my $f (@files) {
+		# If file does not exist (i.e database and file system are not in sync)
+		if(!-e $f) {
+			if($dieOnFileError) {
+				print STDERR "Cannot backup database files:\n";
+				print STDERR "\tFile $f is in the database but was not found in $dataDirBasepath. Aborting.\n";
+				exit $NeuroDB::ExitCodes::MISSING_FILES;
+			}
+			warn "Warning! File $f was not found in $dataDirBasepath (will not be backed up)\n";
+		} elsif(!-r $f) {
+			if($dieOnFileError) {
+				print STDERR "Cannot backup database files:\n";
+				print STDERR "\tFile $f is in database but is not readable. Aborting.\n";
+				exit $NeuroDB::ExitCodes::UNREADABLE_FILE;
+			}
+			warn "Warning! File $f is not readable (will not be backed up)\n";
+		}
+	}
+	
+	# Create a temporary file that will list the absolute paths of all
+	# files to backup (archive). 
+    my($fh, $tmpFileName) = tempfile("$0.filelistXXXX", UNLINK => 1);
+    foreach(@files) {
+		print $fh "$_\n";
+	}
+	close($fh);
+	
+	# Put all files in a big compressed tar ball
+    my $filesBackupPath = &getBackupFileName($uploadID);
+    print "Backing up files related to upload $uploadID...\n";
+	if(system('tar', 'zcvf', $filesBackupPath, '--absolute-names', '--files-from', $tmpFileName)) {
+	    print STDERR "backup command failed: $!\n";
+	    exit $NeuroDB::ExitCodes::PROGRAM_EXECUTION_FAILURE;
+    } 
+
+	print "File $filesBackupPath successfully created.\n";
+}
+
+=pod
+
+=head3 deleteUploadInDatabase($dbh, $uploadID, $tarchiveID, $sessionID, $intermediaryFilesRef, $picFilesRef)
+
+This method deletes all information in the database associated to the given archive. More specifically, it 
+deletes records from tables C<notification_spool>, C<tarchive_files>, C<tarchive_series>, C<files_intermediary>
+C<parameter_file>, C<files>, C<mri_protocol_violated_scans>, C<mri_violations_log>, C<MRICandidateErrors>
+C<mri_upload> and C<tarchive>. It will also set the CScan_done> value of the scan's session to 'N' if the upload
+is the last upload tied to that session. All the delete/update operations are done inside a single transaction so 
+either they all succeed or they all fail (and a rollback is performed).
+
+INPUTS:
+ 
+  - $dbh       : database handle.
+  - $uploadId  : ID of the upload to delete.
+  - $tarchiveID: ID of the tarchive to delete.
+  - $sessionID : ID of the session associated to the scan,
+  - $filesRef: reference to the array that contains all files in table C<files> associated to
+               the upload.
+  - $intermediaryFilesRef: reference to the array that contains all files in table C<files_intermediary>
+                           associated to the upload.
+  - $picFilesRef: reference to the array that contains all files in table C<parameter_file>
+                  associated to the upload.
+                  
+=cut
+sub deleteUploadInDatabase {
+	my($dbh, $uploadID, $tarchiveID, $sessionID, $intermediaryFilesRef, $picFilesRef)= @_;
```

Comment:
all the queries should probably be prepared

Reply:
Already done